packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.tsx

```diff
@@ -0,0 +1,178 @@
+
+// const { unblockUser, createChat } = chatActions
+
+const messages = {
+  title: 'Inbox Unavailable',
+  content: 'You can no longer send messages to this person.',
+  button: 'Learn More',
+  tipContent: (displayName: ReactNode) => (
+    <>
+      {'You must send '}
+      {displayName}
+      {' a tip before you can send them messages.'}
+    </>
+  ),
+  tipButton: 'Send $AUDIO',
+  unblockContent: 'You cannot send messages to users you have blocked.',
+  unblockButton: 'Learn More'
+}
+
+const actionToContent = (
+  action: ChatPermissionAction,
+  user: User,
+  onClose: () => void
+) => {
+  switch (action) {
+    case ChatPermissionAction.NONE:
+      return {
+        content: messages.content,
+        buttonText: messages.button,
+        buttonIcon: null
+      }
+    case ChatPermissionAction.TIP:
+      return {
+        content: messages.tipContent(
+          <UserNameAndBadges user={user} onClickArtistName={onClose} />
+        ),
+        buttonText: messages.tipButton,
+        buttonIcon: <IconTipping />
+      }
+    case ChatPermissionAction.UNBLOCK:
+      return {
+        content: messages.unblockContent,
+        buttonText: messages.unblockButton,
+        buttonIcon: null
+      }
+    default:
+      return {
+        content: messages.content,
+        buttonText: messages.button,
+        buttonIcon: null
+      }
+  }
+}
+
+const { beginTip } = tippingActions
+const { getCanChat } = chatSelectors
+
+const UserNameAndBadges = ({
+  user,
+  onClickArtistName
+}: {
+  user: User
+  onClickArtistName?: () => void
+}) => {
+  const goToRoute = useGoToRoute()
+  const goToProfile = useCallback(() => {
+    goToRoute(profilePage(user.handle))
+    onClickArtistName?.()
+  }, [goToRoute, onClickArtistName, user])
+  return (
+    <ArtistPopover
+      handle={user.handle}
+      component='span'
+      onNavigateAway={onClickArtistName}
+    >
+      <div className={styles.nameAndBadge} onClick={goToProfile}>
+        <span>{user.name}</span>
+        <UserBadges
+          userId={user.user_id}
+          className={styles.badges}
+          badgeSize={14}
+          inline
+        />
+      </div>
+    </ArtistPopover>
+  )
+}
+
+export const InboxUnavailableModal = ({
+  isVisible,
+  onClose,
+  user
+}: {
+  isVisible: boolean
+  onClose: () => void
+  user: User
+}) => {
+  const dispatch = useDispatch()
+  const { callToAction } = useSelector((state) =>
+    getCanChat(state, user.user_id)
+  )
+  const hasAction = callToAction === ChatPermissionAction.TIP
+  // TODO: add `|| callToAction === ChatPermissionAction.UNBLOCK`
+
+  const handleClick = useCallback(() => {
+    if (callToAction === ChatPermissionAction.TIP) {
+      dispatch(beginTip({ user, source: 'inboxUnavailableModal' }))
+      onClose()
+    } else if (callToAction === ChatPermissionAction.UNBLOCK) {
+      // TODO: Pending Design/Product feedback, uncomment this and turn into Unblock button
```

Comment:
Waiting for Design / Product decision here, will remove before merging otherwise